### ISSUE TITLE
Extended Flax NNX/Jax Support

### ISSUE THREAD

**Issue by MauriceFromm** [pwhofman/probly#548]
Epic for extended flax/jax support in probly.

## References

[Flax NNX docs](https://flax.readthedocs.io/en/latest/nnx_basics.html)
[Flax - Randomness (rngs)](https://flax.readthedocs.io/en/latest/guides/randomness.html)
[JAX docs](https://docs.jax.dev/en/latest/)
[JAX - The Sharp Bits (immutable arrays and single-precision default)](https://docs.jax.dev/en/latest/notebooks/Common_Gotchas_in_JAX.html)
[Optax - gradient processing and optimization library for JAX](https://optax.readthedocs.io/en/latest/)

## Out of scope

- `method/laplace` wraps `laplace-torch`, a torch only third party package.
- `method/graph_posterior_network` depends on `torch_geometric`.
